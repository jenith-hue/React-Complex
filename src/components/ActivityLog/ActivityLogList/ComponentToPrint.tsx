import {
  RACTable,
  RACTableCell,
  RACTableRow,
  withStyles,
} from '@rentacenter/racstrap';
import React from 'react';
import clsx from 'clsx';
import {
  DAYS_OF_WEEK,
  formatDate,
  formatDateString,
  formatPhoneNumberForPrintLetter,
  formatStringDateHoursAndMinutes,
} from '../../../utils/utils';
import { AccountActivity } from '../../../domain/ActivityLog/ActivityLogResult';
import { tableHeaderList } from './ActivityLogList';
import logo from './logo.png';
import { StoreDetail, User } from '../../../types/types';
import { WORKED_HISTORY_ACTIVITY_TYPE_CODES } from '../../../constants/constants';

export interface ActivityLogListModalProps {
  activityLog?: AccountActivity[];
  classes?: any;
  selectedDateRangeFromOption: string;
  selectedDateRangeToOption: string;
  store?: StoreDetail;
  user?: User;
}

const useStyles = (theme: any) => ({
  activityLogListWrapper: {
    marginTop: '1rem',
    marginBottom: '7.5rem',
  },
  activityLogList: {
    flexDirection: 'row',
    boxSizing: 'border-box',
    borderRadius: '1rem',
    overflowX: 'auto',
    height: '31.25rem',
    padding: '1rem',
    boxShadow: 'none',
  },
  paddingTopBottom8: {
    paddingTop: '.2rem',
    paddingBottom: '.0rem',
  },
  contentHeight: {
    height: theme.typography.pxToRem(500),
  },
  print: {
    width: '100vw',
  },
  printFontSize: {
    fontSize: theme.typography.pxToRem(8),
    padding: '.2rem .2rem',
  },
  printPreviewHeaderWrapper: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottom: '2px solid Gray',
    padding: theme.typography.pxToRem(10),
    ...theme.typography.body1,
    fontSize: theme.typography.pxToRem(8),
  },
  printPreviewHeaderBlock: {
    display: 'flex',
    flexDirection: 'column',
  },
  printPreviewHeaderCenterBlock: {
    alignSelf: 'center',
    fontSize: theme.typography.pxToRem(10),
  },
  printPreviewHeaderRightBlock: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'end',
    alignSelf: 'center',
    marginTop: theme.typography.pxToRem(70),
  },
  logo: {
    height: theme.typography.pxToRem(60),
    width: theme.typography.pxToRem(60),
    marginBottom: '1rem',
  },
});

// This has to be class component, in order to integrate with react-to-print
class ComponentToPrint extends React.Component<ActivityLogListModalProps> {
  // eslint-disable-next-line sonarjs/cognitive-complexity
  render() {
    const {
      activityLog,
      classes,
      selectedDateRangeFromOption,
      selectedDateRangeToOption,
      store,
      user,
    } = this.props;

    const dateRangeFilter =
      selectedDateRangeFromOption || selectedDateRangeToOption
        ? `${selectedDateRangeFromOption} - ${selectedDateRangeToOption}`
        : 'All';

    const storeAddress = store
      ? `${store.addressLine1}, ${store.city}, ${store.stateName}, ${store.zip}`
      : '';
    const today = new Date();

    const renderTableHead = () => (
      <>
        {tableHeaderList.map((col, index) => (
          <RACTableCell
            key={index}
            classes={{
              root: clsx(classes.paddingTopBottom8, classes.printFontSize),
            }}
          >
            {col}
          </RACTableCell>
        ))}
      </>
    );

    const renderTableContent = () => (
      <>
        {activityLog?.map((log, index) => {
          const code: string | undefined = log.accountActivityType?.code;
          const description: string | undefined =
            code?.toUpperCase() ===
            WORKED_HISTORY_ACTIVITY_TYPE_CODES.CALL_CUSTOMER
              ? log.callResultType?.description
              : log.accountActivityType?.description;
          return (
            <RACTableRow key={index} hover backgroundColor="white">
              <>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {formatDateString(log.activityDate)}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {formatStringDateHoursAndMinutes(log.callDate)}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >{`${log.customerFirstName || ''} ${
                  log.customerLastName || ''
                }`}</RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {log.route?.description}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {log.phoneNumberDialed &&
                    formatPhoneNumberForPrintLetter(log.phoneNumberDialed)}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: clsx(classes.printFontSize),
                  }}
                >
                  {log.phoneContact?.phoneType?.code || ''}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {description || ''}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >{`${log.phoneContact?.firstName || ''} ${
                  log.phoneContact?.lastName || ''
                }`}</RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {log.phoneContact?.relationshipType?.description || ''}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {log.daysLate}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {formatDateString(log.commitment?.commitmentDate)}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {formatStringDateHoursAndMinutes(
                    log.commitment?.commitmentDate
                  )}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {log.commitment?.commitmentStatus?.description || ''}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >
                  {formatDateString(log.pastDueDate)}
                </RACTableCell>
                <RACTableCell
                  classes={{
                    root: classes.printFontSize,
                  }}
                >{`${log.coWorker?.firstName || ''} ${
                  log.coWorker?.lastName || ''
                }`}</RACTableCell>
              </>
            </RACTableRow>
          );
        })}
      </>
    );

    return (
      <div>
        <div className={classes.printPreviewHeaderWrapper}>
          <div className={classes.printPreviewHeaderBlock}>
            <img src={logo} className={classes.logo} alt="logoImage" />
            <span>Rent-A-Center. Inc.</span>
            <span>{`${store?.storeNumber} ${store?.storeName}`}</span>
            <span>{storeAddress}</span>
            <span>For the period {dateRangeFilter}</span>
          </div>
          <div className={classes.printPreviewHeaderCenterBlock}>
            <span>Generate Call Log</span>
          </div>
          <div className={classes.printPreviewHeaderRightBlock}>
            <span>{`${DAYS_OF_WEEK[today.getDay()]} ${formatDate(
              today
            )}`}</span>
            <span>
              Report requested by {`${user?.lastName}, ${user?.firstName}`}
            </span>
          </div>
        </div>
        <RACTable
          renderTableHead={renderTableHead}
          renderTableContent={renderTableContent}
        />
      </div>
    );
  }
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
//@ts-ignore
export default withStyles(useStyles)(ComponentToPrint);
