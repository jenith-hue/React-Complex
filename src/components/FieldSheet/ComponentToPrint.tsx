import React from 'react';
import { FieldSheet } from '../../types/types';
import Template from './Template';

export interface FieldSheetLetterProps {
  fieldSheets?: FieldSheet[];
  classes?: any;
}

// This has to be class component, in order to integrate with react-to-print
class ComponentToPrint extends React.Component<FieldSheetLetterProps> {
  render() {
    const { fieldSheets } = this.props;

    return (
      <div>
        {fieldSheets?.map((sheet: FieldSheet, index: number) => (
          <Template key={index} fieldSheet={sheet} />
        ))}
      </div>
    );
  }
}

export default ComponentToPrint;
