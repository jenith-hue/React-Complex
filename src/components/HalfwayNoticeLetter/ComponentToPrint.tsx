import { Typography, withStyles } from '@rentacenter/racstrap';
import clsx from 'clsx';
import React from 'react';
import { getNameForAbbreviation, StateAbbreviation } from '../../domain/Common';
import { PrintLetterDetails } from '../../types/types';
import {
  formatDateString,
  formatPhoneNumberForPrintLetter,
} from '../../utils/utils';

export interface HalfwayNoticeLetterProps {
  printLetterDetails: PrintLetterDetails;
  classes?: any;
}

const useStyles = (theme: any) => ({
  allLettersContainer: {
    width: '100%',
  },
  breakPage: {
    breakBefore: 'page',
  },
  halfwayNoticeContainer: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    padding: theme.typography.pxToRem(30),
    background: 'white',
  },
  section: {
    paddingBottom: theme.typography.pxToRem(30),
  },
  paragraph: {
    paddingBottom: theme.typography.pxToRem(15),
  },
  headerMetadata: {
    width: theme.typography.pxToRem(400),
  },
  underline: {
    textDecoration: 'underline',
  },
  alignLeft: {
    textAlign: 'left',
  },
  lineHeight7: {
    lineHeight: theme.typography.pxToRem(7),
  },
  margin20: {
    marginBottom: theme.typography.pxToRem(20),
  },
});

const dateFormat = 'MM-dd-yyyy';

// This has to be class component, in order to integrate with react-to-print
class ComponentToPrint extends React.Component<HalfwayNoticeLetterProps> {
  render() {
    const {
      classes,
      printLetterDetails: { customer, store, dateOfNotice },
    } = this.props;
    return (
      <div className={classes.allLettersContainer}>
        {customer.map(
          (
            {
              firstName,
              lastName,
              rentalAgreementExpiration,
              deadlineToReturnProperty,
              address: customerAddress,
            },
            index
          ) => {
            const { address, phoneNumber } = store;
            const storeAddressLines = [
              address.addressLine1,
              address.addressLine2,
            ]
              .filter(Boolean)
              .join(', ');
            const storeAddressDetails = [
              address.city,
              getNameForAbbreviation(address.state as StateAbbreviation),
              address.zipCode,
            ]
              .filter(Boolean)
              .join(', ');
            const formatedPhoneNumber =
              formatPhoneNumberForPrintLetter(phoneNumber) || phoneNumber;
            const customerName = `${firstName} ${lastName}`;
            const customerAddressLineDetails = [
              customerAddress.addressLine1,
              customerAddress.addressLine2,
            ]
              .filter(Boolean)
              .join(', ');
            const customerAddressDetails = [
              customerAddress.city,
              getNameForAbbreviation(
                customerAddress.state as StateAbbreviation
              ),
              customerAddress.zipCode,
            ]
              .filter(Boolean)
              .join(', ');

            return (
              <span key={index}>
                <div className={classes.halfwayNoticeContainer}>
                  <div className={clsx(classes.section, classes.alignLeft)}>
                    <Typography align="left" variant="h6">
                      Rent-A-Center, Inc.
                    </Typography>
                  </div>
                  <div
                    className={clsx(classes.section, classes.headerMetadata)}
                  >
                    <Typography
                      align="left"
                      variant="h6"
                      className={clsx(classes.paragraph, classes.lineHeight7)}
                    >
                      {storeAddressLines}
                    </Typography>
                    <Typography
                      align="left"
                      variant="h6"
                      className={clsx(classes.paragraph, classes.lineHeight7)}
                    >
                      {storeAddressDetails}
                    </Typography>
                    <Typography
                      align="left"
                      variant="h6"
                      className={clsx(
                        classes.paragraph,
                        classes.lineHeight7,
                        classes.margin20
                      )}
                    >
                      {formatedPhoneNumber}
                    </Typography>
                    <Typography
                      align="left"
                      variant="h6"
                      className={clsx(classes.paragraph, classes.lineHeight7)}
                    >
                      {customerName}
                    </Typography>
                    <Typography
                      align="left"
                      variant="h6"
                      className={clsx(classes.paragraph, classes.lineHeight7)}
                    >
                      {customerAddressLineDetails}
                    </Typography>
                    <Typography
                      align="left"
                      variant="h6"
                      className={clsx(classes.paragraph, classes.lineHeight7)}
                    >
                      {customerAddressDetails}
                    </Typography>
                  </div>
                  <div className={clsx(classes.section, classes.alignLeft)}>
                    <Typography
                      variant="h5"
                      align="center"
                      className={classes.paragraph}
                    >
                      5-Day Demand Notice For Return of Rented Property
                    </Typography>
                    <div>
                      <Typography align="left" display="inline" variant="h5">
                        Date Rental Agreement Expired:
                      </Typography>
                      <Typography
                        align="left"
                        display="inline"
                        variant="h6"
                        className={classes.underline}
                      >
                        {` ${formatDateString(
                          rentalAgreementExpiration.split('T')[0],
                          dateFormat
                        )}`}
                      </Typography>
                    </div>
                    <div>
                      <Typography align="left" display="inline" variant="h5">
                        Date of Notice:
                      </Typography>
                      <Typography
                        align="left"
                        display="inline"
                        variant="h6"
                        className={classes.underline}
                      >
                        {` ${formatDateString(
                          dateOfNotice.split('T')[0],
                          dateFormat
                        )}`}
                      </Typography>
                    </div>
                    <div>
                      <Typography align="left" display="inline" variant="h5">
                        Deadline to Return Property:
                      </Typography>
                      <Typography
                        align="left"
                        display="inline"
                        variant="h6"
                        className={classes.underline}
                      >
                        {` ${formatDateString(
                          deadlineToReturnProperty.split('T')[0],
                          dateFormat
                        )} `}
                      </Typography>
                      <Typography align="left" display="inline" variant="h6">
                        ( if a Sunday or a holiday, then the next business day).
                      </Typography>
                    </div>
                    <div className={classes.paragraph}>
                      <Typography align="left" display="inline" variant="h6">
                        To:
                      </Typography>
                      <Typography
                        align="left"
                        display="inline"
                        variant="h6"
                        className={classes.underline}
                      >
                        {` ${customerName}`}
                      </Typography>
                    </div>
                    <div className={classes.paragraph}>
                      <Typography align="left" variant="h6">
                        Your rental agreement expired more than 15 days ago when
                        you failed to make a renewal payment. Nevertheless, you
                        have refused to either renew your agreement or make
                        arrangements to return our property to our store.
                      </Typography>
                    </div>
                    <div className={classes.paragraph}>
                      <Typography align="left" variant="h6">
                        You must either return the rented property to our store
                        or contact me at the above phone number to schedule a
                        pick up time. We must receive the property no later than
                        5:00 p.m.on the above deadline date. This is consistent
                        with our legal rights under your rental agreement.
                      </Typography>
                    </div>
                    <div className={classes.paragraph}>
                      <Typography align="left" variant="h6">
                        Do not underestimate the seriousness of this situation.
                        Please call the Store Manager at the above phone number
                        to discuss the many options we have available to
                        successfully resolve this matter. We anticipate your
                        visit or call soon.
                      </Typography>
                    </div>
                  </div>
                  <div className={classes.paragraph}>
                    <Typography align="left" variant="h6">
                      Sincerely,
                    </Typography>
                    <Typography
                      align="left"
                      variant="h6"
                      className={classes.underline}
                    >
                      {/* {storeManagerName} */}
                    </Typography>
                    <Typography align="left" variant="h6">
                      Store Manager
                    </Typography>
                  </div>
                </div>
                <div className={classes.breakPage}> </div>
              </span>
            );
          }
        )}
      </div>
    );
  }
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
//@ts-ignore
export default withStyles(useStyles)(ComponentToPrint);
