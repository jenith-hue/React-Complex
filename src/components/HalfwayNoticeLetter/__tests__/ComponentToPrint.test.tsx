/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-identical-functions */
import React from 'react';

import renderer from 'react-test-renderer';
import ComponentToPrint from '../ComponentToPrint';

interface PropWithChildren {
  children: JSX.Element;
}

jest.mock('@rentacenter/racstrap', () => ({
  Typography: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Typography" {...rest}>
      {children}
    </div>
  ),
  RACCOLOR: {
    ALICE_BLUE: 'ALICE_BLUE',
  },
  makeStyles: () => () => ({
    allLettersContainer: 'allLettersContainer',
    breakPage: 'breakPage',
    halfwayNoticeContainer: 'halfwayNoticeContainer',
    section: 'section',
    paragraph: 'paragraph',
    headerMetadata: 'headerMetadata',
    underline: 'underline',
    alignLeft: 'alignLeft',
  }),
  withStyles: () => (Component: any) => Component,
}));

const classes = {
  allLettersContainer: 'allLettersContainer',
  breakPage: 'breakPage',
  halfwayNoticeContainer: 'halfwayNoticeContainer',
  section: 'section',
  paragraph: 'paragraph',
  headerMetadata: 'headerMetadata',
  underline: 'underline',
  alignLeft: 'alignLeft',
};
describe('ComponentToPrint', () => {
  it('renders correctly, when multiple letters are present', () => {
    const tree = renderer.create(
      <ComponentToPrint
        classes={classes}
        printLetterDetails={{
          letterType: 'COL',
          dateOfNotice: '2021-08-22',
          customer: [
            {
              customerId: 123456,
              globalCustomerId: '1fc287ca-b2e8-4829-aed4-8a5552f7d048',
              firstName: 'Marissa',
              lastName: 'Horn',
              address: {
                addressLine1: '18777 Midway RD',
                addressLine2: 'APT 401',
                city: 'DALLAS',
                state: 'TX',
                zipCode: '75287-2715',
              },
              rentalAgreementExpiration: '2021-08-12',
              deadlineToReturnProperty: '2021-08-27',
            },
          ],
          store: {
            storeNumber: '04744',
            name: 'DALLAS-BUCKNER BLVD',
            address: {
              addressLine1: '1322 S BUCKNER BLVD',
              addressLine2: '',
              city: 'DALLAS',
              state: 'TX',
              zipCode: '75217-1701',
            },
            phoneNumber: '2143985174',
          },
        }}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('renders correctly, when letter array is empty', () => {
    const tree = renderer.create(
      <ComponentToPrint
        printLetterDetails={{
          letterType: 'COL',
          dateOfNotice: '2021-08-22',
          customer: [],
          store: {
            storeNumber: '04744',
            name: 'DALLAS-BUCKNER BLVD',
            address: {
              addressLine1: '1322 S BUCKNER BLVD',
              addressLine2: '',
              city: 'DALLAS',
              state: 'TX',
              zipCode: '75217-1701',
            },
            phoneNumber: '2143985174',
          },
        }}
        classes={classes}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
