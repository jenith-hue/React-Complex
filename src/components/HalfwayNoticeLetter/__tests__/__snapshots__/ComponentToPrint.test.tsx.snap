// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentToPrint renders correctly, when letter array is empty 1`] = `
<div
  className="allLettersContainer"
/>
`;

exports[`ComponentToPrint renders correctly, when multiple letters are present 1`] = `
<div
  className="allLettersContainer"
>
  <span>
    <div
      className="halfwayNoticeContainer"
    >
      <div
        className="section alignLeft"
      >
        <div
          align="left"
          data-testid="Typography"
          variant="h6"
        >
          Rent-A-Center, Inc.
        </div>
      </div>
      <div
        className="section headerMetadata"
      >
        <div
          align="left"
          className="paragraph"
          data-testid="Typography"
          variant="h6"
        >
          1322 S BUCKNER BLVD
        </div>
        <div
          align="left"
          className="paragraph"
          data-testid="Typography"
          variant="h6"
        >
          DALLAS, Texas, 75217-1701
        </div>
        <div
          align="left"
          className="paragraph"
          data-testid="Typography"
          variant="h6"
        >
          (214) 398-5174
        </div>
        <div
          align="left"
          className="paragraph"
          data-testid="Typography"
          variant="h6"
        >
          Marissa Horn
        </div>
        <div
          align="left"
          className="paragraph"
          data-testid="Typography"
          variant="h6"
        >
          18777 Midway RD, APT 401
        </div>
        <div
          align="left"
          className="paragraph"
          data-testid="Typography"
          variant="h6"
        >
          DALLAS, Texas, 75287-2715
        </div>
      </div>
      <div
        className="section alignLeft"
      >
        <div
          align="center"
          className="paragraph"
          data-testid="Typography"
          variant="h5"
        >
          5-Day Demand Notice For Return of Rented Property
        </div>
        <div>
          <div
            align="left"
            data-testid="Typography"
            display="inline"
            variant="h5"
          >
            Date Rental Agreement Expired:
          </div>
          <div
            align="left"
            className="underline"
            data-testid="Typography"
            display="inline"
            variant="h6"
          >
             08-12-2021
          </div>
        </div>
        <div>
          <div
            align="left"
            data-testid="Typography"
            display="inline"
            variant="h5"
          >
            Date of Notice:
          </div>
          <div
            align="left"
            className="underline"
            data-testid="Typography"
            display="inline"
            variant="h6"
          >
             08-22-2021
          </div>
        </div>
        <div>
          <div
            align="left"
            data-testid="Typography"
            display="inline"
            variant="h5"
          >
            Deadline to Return Property:
          </div>
          <div
            align="left"
            className="underline"
            data-testid="Typography"
            display="inline"
            variant="h6"
          >
             08-27-2021 
          </div>
          <div
            align="left"
            data-testid="Typography"
            display="inline"
            variant="h6"
          >
            ( if a Sunday or a holiday, then the next business day).
          </div>
        </div>
        <div
          className="paragraph"
        >
          <div
            align="left"
            data-testid="Typography"
            display="inline"
            variant="h6"
          >
            To:
          </div>
          <div
            align="left"
            className="underline"
            data-testid="Typography"
            display="inline"
            variant="h6"
          >
             Marissa Horn
          </div>
        </div>
        <div
          className="paragraph"
        >
          <div
            align="left"
            data-testid="Typography"
            variant="h6"
          >
            Your rental agreement expired more than 15 days ago when you failed to make a renewal payment. Nevertheless, you have refused to either renew your agreement or make arrangements to return our property to our store.
          </div>
        </div>
        <div
          className="paragraph"
        >
          <div
            align="left"
            data-testid="Typography"
            variant="h6"
          >
            You must either return the rented property to our store or contact me at the above phone number to schedule a pick up time. We must receive the property no later than 5:00 p.m.on the above deadline date. This is consistent with our legal rights under your rental agreement.
          </div>
        </div>
        <div
          className="paragraph"
        >
          <div
            align="left"
            data-testid="Typography"
            variant="h6"
          >
            Do not underestimate the seriousness of this situation. Please call the Store Manager at the above phone number to discuss the many options we have available to successfully resolve this matter. We anticipate your visit or call soon.
          </div>
        </div>
      </div>
      <div
        className="paragraph"
      >
        <div
          align="left"
          data-testid="Typography"
          variant="h6"
        >
          Sincerely,
        </div>
        <div
          align="left"
          className="underline"
          data-testid="Typography"
          variant="h6"
        />
        <div
          align="left"
          data-testid="Typography"
          variant="h6"
        >
          Store Manager
        </div>
      </div>
    </div>
    <div
      className="breakPage"
    >
       
    </div>
  </span>
</div>
`;
